implementation/src/main/java/io/smallrye/config/ConfigMappingLoader.java

```diff
@@ -81,7 +81,7 @@ static ConfigMappingInterface getConfigMappingInterface(final Class<?> type) {
         return (Class<? extends ConfigMappingObject>) loadClass(type, mappingMetadata);
     }
 
-    static Class<?> loadClass(final Class<?> parent, final ConfigMappingMetadata configMappingMetadata) {
+    static synchronized Class<?> loadClass(final Class<?> parent, final ConfigMappingMetadata configMappingMetadata) {
```

Comment:
While it should be mostly fine, since I believe applications don't ship with a huge number of mapping classes, we could probably improve this with a lock to the class name instead (like the `Classloader`). Check https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/lang/ClassLoader.java#L663.

Do you mind updating the PR? Thanks!

Reply:
Thanks for the feedback, PR has been updated accordingly.